src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -129,16 +131,40 @@ export default function withAnnotations<P extends object>(
             }
         };
 
+        handleAnnotationsFetchError = ({ error }: { error: Error }) => {
+            this.fetchError = error;
+        };
+
         handleAnnotator = (annotator: Annotator): void => {
             this.annotator = annotator;
             this.annotator.addListener('annotatorevent', this.handleAnnotatorEvent);
+
+            if (this.fetchError) {
+                this.handleSidebarFetchError(this.fetchError);
+            }
+        };
+
+        handleAnnotatorCreate = (annotator: Annotator): void => {
+            annotator.addListener('annotations_fetch_error', this.handleAnnotationsFetchError);
+        };
+
+        handleSidebarFetchError = (error: Error) => {
+            const { annotator } = this;
+
+            if (!annotator) {
+                this.fetchError = error;
```

Comment:
We should be tracking state in appropriate ways. Changes to instance properties will not cause a re-render, even though this value is used in render.